File: 0029-divide-two-integers/0029-divide-two-integers.c
int divide(int dividend, int divisor) {
    double ans = (double)dividend/divisor;
    if(ans<(pow(2,31)-1) && ans>(pow(-2,31)))
        return trunc(dividend/divisor);
    else if(ans>=(pow(2,31))-1)
        return pow(2,31)-1;
    return pow(-2,31);
}

File: 0029-divide-two-integers/0029-divide-two-integers.java
class Solution {
    public int divide(int dividend, int divisor) {
        double ans = (double)dividend/divisor;
        if (ans > Integer.MAX_VALUE) 
            return Integer.MAX_VALUE;
        else if (ans < Integer.MIN_VALUE) 
            return Integer.MIN_VALUE;
    return (int) ans; 
    }
}